### ISSUE TITLE
Bug: LogRecord parsing fails when pytest captures logs with extra fields

### ISSUE THREAD

**Issue by MarcusJellinghaus** [MarcusJellinghaus/mcp-code-checker#74]
## Summary

When running `run_pytest_check` on projects that use Python's logging `extra` parameter, the JSON report parsing fails with:
```
Error running pytest: LogRecord.__init__() got an unexpected keyword argument 'package_name'
```

## Root Cause (Confirmed)

The `LogRecord` dataclass in `models.py` has a fixed set of standard Python logging fields. However, `pytest-json-report` captures the **entire** `LogRecord.__dict__`, which includes any custom fields added via the `extra` parameter.

**Code flow:**
1. Target project: `logger.debug("msg", extra={"package_name": "my_package"})`
2. Python logging adds `package_name` to `LogRecord.__dict__`
3. pytest-json-report's `LoggingHandler.emit()` captures `dict(record.__dict__)` → includes `package_name`
4. mcp-code-checker's `parsers.py:44`: `LogRecord(**log_record_data)` → **fails on unexpected field**

## Implementation Plan

**Approach:** Preserve extra fields and apply defensive parsing

**Changes:**

1. **`models.py`** - Add `extra: Dict[str, Any]` field to `LogRecord`:
   ```python
   extra: Dict[str, Any] = field(default_factory=dict)
   ```

2. **`parsers.py`** - Create helper function for defensive parsing:
   - Filter known fields for dataclass constructor
   - Capture unknown fields in `extra` dict (for LogRecord)
   - Apply to all dataclasses: `LogRecord`, `Crash`, `TracebackEntry`, `Warning`, `CollectorResult`

## Affected Files

- `src/mcp_code_checker/code_checker_pytest/models.py`
- `src/mcp_code_checker/code_checker_pytest/parsers.py`

## Workaround

Pass `--json-report-omit log` to pytest to skip log capture:
```python
mcp__code-checker__run_pytest_check(extra_args=["--json-report-omit", "log"])
```

---

# Original issue: Bug: LogRecord.__init__() got an unexpected keyword argument 'package_name' when project uses extra parameter in logging

## Description

When running `run_pytest_check` on a project that uses Python's standard logging with the `extra` parameter containing a `package_name` key, the MCP server fails with:

```
Error running pytest: LogRecord.__init__() got an unexpected keyword argument 'package_name'
```

## Steps to Reproduce

1. Have a project with code that uses standard Python logging with `extra` parameter:
   ```python
   import logging
   logger = logging.getLogger(__name__)
   
   logger.debug(
       "Looking for data file",
       extra={"package_name": "my_package", "relative_path": "file.txt"},
   )
   ```

2. Run the MCP pytest check tool:
   ```python
   mcp__code-checker__run_pytest_check(extra_args=["-n", "auto"])
   ```

3. The tool returns the error instead of running tests.

## Expected Behavior

Tests should run successfully. The `extra` parameter in Python's logging is designed to add arbitrary key-value pairs to log records and should not conflict with LogRecord initialization.

## Actual Behavior

The MCP server crashes with `LogRecord.__init__() got an unexpected keyword argument 'package_name'`.

## Workaround

Running pytest directly from command line works fine:
```bash
python -m pytest -n auto -m "not integration"
# Result: 1397 passed in 21.89s
```

## Root Cause Analysis

The issue appears to be in how the mcp-code-checker configures logging. Possible causes:

1. **Custom LogRecord factory**: If `logging.setLogRecordFactory()` is used with a custom class that doesn't accept `**kwargs` or doesn't handle extra fields properly
2. **Logging configuration conflict**: The MCP server's logging setup may be interfering with the target project's logging calls
3. **python-json-logger interaction**: Both mcp-code-checker and the target project use `python-json-logger` - there may be a configuration conflict

## Environment

- **mcp-code-checker version**: 0.1.1
- **Python version**: 3.13.0
- **OS**: Windows 11
- **Dependencies**: python-json-logger, structlog, pytest, pytest-xdist

## Additional Context

The target project (`mcp_coder`) uses standard Python logging patterns:
```python
# In src/mcp_coder/utils/data_files.py
logger.debug(
    "SEARCH STARTED: Looking for data file",
    extra={
        "package_name": package_name,
        "relative_path": relative_path,
    },
)
```

This is valid Python logging - the `extra` dict should be merged into the LogRecord's `__dict__` after initialization, not passed as keyword arguments to `__init__`.
